### PR TITLE
add Vigenere Cipher

### PR DESCRIPTION
**Pull request by Artyko**
I'm new in contribution. May you give me feedback on my PR plz?

### CODE REVIEW

File: encryptor/__init__.py [hunk 1]
```diff
@@ -66,6 +66,24 @@ def ceasar_decrypt(text):
             "content": Ceasar.decrypt(text, offset)
         })
 
+    @app.route("/api/vigenere/encrypt/<text>")
+    def vigenere_encrypt(text):
+        offset = request.args.get("tabula_recta", Vigenere.ALPHABET)
```
Comment (rafaeelaudibert) [inline encryptor/__init__.py]: This should probably be named `tabula_recta`

File: encryptor/__init__.py [hunk 2]
```diff
@@ -66,6 +66,24 @@ def ceasar_decrypt(text):
...
+
+        return jsonify({
+            "status": 200,
+            "content": Vigenere.encrypt(text, offset)
```
Comment (rafaeelaudibert) [inline encryptor/__init__.py]: After renaming the parameter as mentioned above, the `key` parameter is also missing. It should probably be parsed the same way as the `tabula_recta`

File: encryptor/__init__.py [hunk 3]
```diff
@@ -66,6 +66,24 @@ def ceasar_decrypt(text):
...
+        })
+
+    @app.route("/api/vigenere/decrypt/<text>")
+    def vigenere_decrypt(text):
+        offset = request.args.get("tabula_recta", Vigenere.ALPHABET)
+
+        return jsonify({
+            "status": 200,
+            "content": Vigenere.decrypt(text, offset)
```
Comment (rafaeelaudibert) [inline encryptor/__init__.py]: Same as above applies here. Variable renaming, and missing `key` parameter

File: encryptor/encryptors/vigenere.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    @staticmethod
+    def encrypt(text: str, key: str, tabula_recta=None):
+        if tabula_recta is None:
+            tabula_recta = Vigenere.ALPHABET
+        message_encoded = Vigenere._encode(text)
+        key_encoded = Vigenere._encode(key)
+        compare = Vigenere._comparator(message_encoded, key_encoded)
+        encrypt_message = [(value[0] + value[1]) % len(Vigenere.ALPHABET) for value in compare.values()]
+        return ''.join(Vigenere._decode(encrypt_message))
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/vigenere.py]: Should this really be `_decode`? I'm not sure

Comment (Artyko) [reply encryptor/encryptors/vigenere.py]: Yes, it's decode from ASCII table to char

Comment (Artyko) [reply encryptor/encryptors/vigenere.py]: Yes, it's decode from ASCII table to char

Comment (rafaeelaudibert) [reply encryptor/encryptors/vigenere.py]: Okay!

**Comment by Artyko** [comment]
I'm working on it

File: encryptor/encryptors/vigenere.py [hunk 5]
```diff
@@ -6,26 +7,33 @@ class Vigenere:
...
-        encrypt_message = [(value[0] + value[1]) % len(Vigenere.ALPHABET) for value in compare.values()]
-        return ''.join(Vigenere._decode(encrypt_message))
+        if key.isalpha():
+            message_encoded = Vigenere._encode(text)
+            key_encoded = Vigenere._encode(key)
+            compare = Vigenere._comparator(message_encoded, key_encoded)
+            encrypt_message = [(value[0] + value[1]) % len(Vigenere.ALPHABET) for value in compare.values()]
+            return ''.join(Vigenere._decode(encrypt_message))
+        else:
+            raise WrongTypeParameter("key", str, type(key))
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/vigenere.py]: I think the NotAllowedValue would be a better error here

File: encryptor/encryptors/vigenere.py [hunk 6]
```diff
@@ -6,26 +7,33 @@ class Vigenere:
...
-        decrypt_message = [(value[0] - value[1]) % len(Vigenere.ALPHABET) for value in compare.values()]
-        return ''.join(Vigenere._decode(decrypt_message))
+        if key.isalpha():
+            message_decoded = Vigenere._encode(text)
+            key_decoded = Vigenere._encode(key)
+            compare = Vigenere._comparator(message_decoded, key_decoded)
+            decrypt_message = [(value[0] - value[1]) % len(Vigenere.ALPHABET) for value in compare.values()]
+            return ''.join(Vigenere._decode(decrypt_message))
+        else:
+            raise WrongTypeParameter("key", str, type(key))
```
Comment (rafaeelaudibert) [inline encryptor/encryptors/vigenere.py]: I think the NotAllowedValue would be a better error here

File: tests/api/api_vigenere_test.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    response = client.get("/api/vigenere/decrypt/encryptthis").get_json()
+    assert response["status"] == 200
+    assert response["content"] == "EGUSZRfUIKe"
+
+
+def test_vigenere_encrypt_invalid_key(client):
+    response = client.get(
+        "/api/vigenere/encrypt/encryptthis?key=5").get_json()
+    assert response["status"] == 400
+    assert "Should be <class 'str'>" in response["content"]
```
Comment (rafaeelaudibert) [inline tests/api/api_vigenere_test.py]: When changing the error type, make sure to change the error string detection

**Comment by rafaeelaudibert** [comment]
After the reviews I've commented above, I think we are done!

File: tests/api/api_vigenere_test.py (hunk 7 above)
Comment (Artyko) [reply tests/api/api_vigenere_test.py]: Key must be a 'str' type, and I change it in the error string

Comment (rafaeelaudibert) [reply tests/api/api_vigenere_test.py]: Yes, but the `NotAllowedValue` doesn't return this as the string error. It is enough for you to check only `response["status"]`.

Comment (Artyko) [reply tests/api/api_vigenere_test.py]: Oh, I see

**Comment by rafaeelaudibert** [comment]
Thanks for the contribution, @Artyko! I'm glad you dedicated some of your time to this project. Hope you have a great Hacktoberfest!

**Pull request merged**